### ISSUE TITLE
Rage mechanic issue.

### ISSUE THREAD

**Issue by Gendori** [dalaranwow/dalaran-wow#1254]
Hello, so. Rage mechanic for Warrior is kind of buggy. So far, the only ways for a Warrior to gain Rage are: Anger Management, the Arms talent that grants Rage while in combat. Shield Specialization for Protection, that grants 5 rage on Block/Dodge/Parry. Hitting the mobs/player with auto-attack. There's something missing here, isn't there? There is. Being struck in combat leaves you with no rage generation. I've tested, and tested this while leveling. You gain one rage a second. One rage per second. If you don't gain rage when hit? Means Tanks are rage starved, and have problems tanking. This is mainly a tanking issue, as it makes it much, much more difficult to tank when you only generate rage when auto-attacking, or Shield Specialization. Other people can attest to this. I can attest to this. It just kind of makes Warriors that much more difficult, honestly.

To sum it up: One rage a second, if that. No rage when struck. 

Edit: Druids also share this same problem, so if they go Bear form, to tank something, they won't gain rage when struck.

**Comment by NLJamie** [comment]
Some great bugfixing here, over a year later. Nice job guys /s

**Comment by taralej90** [comment]
Hello!

While this issue has been fixed, I'd like to draw attention to another rage-generation problem:

Warriors (and bears?) do not get the correct amount of rage from attacking target dummies. This makes DPS tests for Warriors pretty much impossible.

wowwiki link, including formulae and relevant for WotLK:
http://www.wowwiki.com/Rage

**Comment by luledebule** [comment]
this issue has not been fixed i dont know why you'd think otherwise have you even tested it?
this also makes the core ability Berserker Rage useless in PvE.

Berserker Rage in WotlK
-The warrior enters a berserker rage, removing and granting immunity to Fear, Sap and Incapacitate effects and **generating extra rage when taking damage.** Lasts 10 sec.

According to the tooltip you're supposed to generate extra rage when taking damage while here on Dalaran-WoW youre not generating rage while being hit period. I recall back in retail i used this all the time when engaging bosses to get early threat and as part of my rotation in general and i am really dissapointed by this.

**Comment by NLJamie** [comment]
Yep it's completely game breaking from a warrior point of view really, count me out from ever playing here again. Can't believe it isn't fixed over a year later.

**Comment by taralej90** [comment]
You do generate rage from damage taken. How else do you think Warriors can tank? I also tested Berserker Rage and I did get extra rage from taking damage.

My test: I went under Dalaran and started playing with the Sinewy Wolves down there. They hit for 160-200, which is somewhere around the amount needed to generate 1 rage point. When I used Berserker Rage I would gain 2 rage from being hit. The same also applies for bosses - it's a simple formula of the damage you've taken and your level conversing into 1 rage where Berserker Rage reduces the amount needed for 1 rage.

You say it's really easy to test, then what kind of tests did you do?

**Comment by MrNaga** [comment]
I can confirm that this is still an issue as of today, april 28th, 2016.
I have been playing a warrior for a week now, tank and dd. I don't think I've ever crossed the 50 rage treshold!
Although it may work for some warriors (like taralej90), for others it does not and it really downgrades the entire experience. Instead of button mashing it is snoozing in front of your keyboard until you have finally enough rage to do something. That's no fun. :(
This mechanic is so central to the warrior gameplay that I believe it should get a high priority.

**Comment by rbccrew** [comment]
Why is this on hold for 2 years now? This makes leveling a warrior so much harder, especially when you roll a prot for RDF leveling... Please fix this, I don't want to abandon my warrior.

**Comment by robinsch** [comment]
Please reopen a valid ticket, a 2 years old ticket with bugs that are fixed or not? doesn't help anyone. If you state what exactly is bugged and add claims that state the bug I will fix it within a day.
